callouts/java/service-callout/src/main/java/service/ServiceCalloutTools.java

```diff
@@ -0,0 +1,111 @@
+import io.envoyproxy.envoy.service.ext_proc.v3.CommonResponse;
+import io.envoyproxy.envoy.service.ext_proc.v3.HeaderMutation;
+import io.envoyproxy.envoy.service.ext_proc.v3.HeadersResponse;
+
+import java.util.Map;
+
+
+/**
+ * ServiceCalloutTools provides utility methods for handling HTTP header and body mutations in service callouts.
+ */
+public class ServiceCalloutTools {
+
+    /**
+     * Adds header mutations to the response builder.
+     * @param headersResponseBuilder Builder for modifying response headers
+     * @param add Iterable containing header key-value pairs to be added
+     */
+    public static void AddHeaderMutations(
+            HeadersResponse.Builder headersResponseBuilder, Iterable<Map.Entry<String, String>> add) {
+        if (add != null) {
+            HeaderMutation.Builder headerMutationBuilder =
+                    headersResponseBuilder.getResponseBuilder().getHeaderMutationBuilder();
+            for (Map.Entry<String, String> entry : add) {
+                headerMutationBuilder
+                        .addSetHeadersBuilder()
+                        .getHeaderBuilder()
+                        .setKey(entry.getKey())
+                        .setValue(entry.getValue())
+                        .setRawValue(ByteString.copyFromUtf8(entry.getValue()));
+            }
+        }
+    }
+
+    /**
+     * Configures the headers response.
+     * @param headersResponseBuilder Builder for modifying response headers
+     * @param add Iterable containing header value options to be added
+     * @param remove Iterable containing header keys to be removed
+     * @param clearRouteCache Boolean indicating whether to clear the route cache
+     * @return Constructed HeadersResponse object
+     */
+    public static HeadersResponse ConfigureHeadersResponse(
+            HeadersResponse.Builder headersResponseBuilder,
+            Iterable<HeaderValueOption> add,
+            Iterable<String> remove,
+            Boolean clearRouteCache) {
+        CommonResponse.Builder responseBuilder = headersResponseBuilder.getResponseBuilder();
+        HeaderMutation.Builder headerBuilder = responseBuilder.getHeaderMutationBuilder();
+        if (add != null) {
+            headerBuilder.addAllSetHeaders(add);
+        }
+        if (remove != null) {
+            headerBuilder.addAllRemoveHeaders(remove);
+        }
+        if (clearRouteCache != null) {
+            responseBuilder.setClearRouteCache(clearRouteCache);
+        }
+        return headersResponseBuilder.build();
+    }
+
+    /**
+     * Builds a body mutation response.
+     * @param bodyResponseBuilder Builder for modifying response body
+     * @param body The body content to be set
+     * @param clearBody Boolean indicating whether to clear the body
+     * @param clearRouteCache Boolean indicating whether to clear the route cache
+     * @return Constructed BodyResponse object
+     */
+    public static BodyResponse BuildBodyMutationResponse(
```

Comment:
Can we rename this to `AddBodyMutations`

Reply:
Done